ornl_nitrogen_setup.sh

```diff
@@ -0,0 +1,155 @@
+spack compilers
+echo --- Modules list
+module list
+echo --- End listings
+
+#
+# Versions should be consistent with setup script
+#
+
+# GCC
+# Dates at https://gcc.gnu.org/releases.html
+gcc_vnew=9.2.0 # 2019-08-12
+
+#Zen2 optimziations are only in gcc 9.1+, with improved scheduling in 9.2+
+#For now, only use newer compilers
+
+# For CUDA toolkit compatibility
+gcc_vcuda=8.3.0 #  2019-02-22
+
+# LLVM 
+# Dates at http://releases.llvm.org/
+llvm_vnew=9.0.0 # 2019-09-19
+#Zen2 scheduleing optimization are only ing LLVM 10+
+
+# HDF5
+hdf5_vnew=1.10.5 # Releeased 2019-02-28
+hdf5_vold=1.8.19 # Released 2017-06-16
+
+# CMake 
+# Dates at https://cmake.org/files/
+cmake_vnew=3.16.2 # Released 2019-12-19
+cmake_vold=3.10.2 # Released 2018-01-18
+
+# OpenMPI
+# Dates at https://www.open-mpi.org/software/ompi/v4.0/
+ompi_vnew=4.0.2 # Released 2019-10-07
+ompi_vold=2.1.2 # Released 2017-09-20
+
+libxml2_vnew=2.9.9 # Released 2019-01-03 See http://xmlsoft.org/sources/
+libxml2_vold=2.9.1 # Released 2013-04-19
+
+# FFTW
+# Dates at http://www.fftw.org/release-notes.html
+fftw_vnew=3.3.8 # Released 2018-05-28
+fftw_vold=3.3.4 # Released 2014-03-16
+
+# BOOST
+# Dates at https://www.boost.org/users/history/
+boost_vnew=1.70.0 # Released 2019-04-12
+boost_vold=1.65.1 # Released 2016-05-13
+
+echo --- START env `date`
+echo --- gcc@${gcc_vnew}
+spack install gcc@${gcc_vnew}
+echo --- load gcc@${gcc_vnew}
+spack load gcc@${gcc_vnew}
+module list
+spack compiler find
+spack install libxml2@${libxml2_vnew}%gcc@${gcc_vnew}
+spack install cmake@${cmake_vnew}%gcc@${gcc_vnew}
+spack install boost@${boost_vnew}%gcc@${gcc_vnew}
+spack install openmpi@${ompi_vnew}%gcc@${gcc_vnew} fabrics=auto ^libxml2@${libxml2_vnew}%gcc@${gcc_vnew}
+#spack HDF5 package requires fortran and hl (high level) support to be specifically enabled for use with QE
+spack install hdf5@${hdf5_vnew}%gcc@${gcc_vnew} +fortran +hl ^openmpi@${ompi_vnew}%gcc@${gcc_vnew} fabrics=auto
+spack install fftw@${fftw_vnew}%gcc@${gcc_vnew} ^openmpi@${ompi_vnew}%gcc@${gcc_vnew} fabrics=auto
+spack unload gcc@${gcc_vnew}
+echo --- gcc@${gcc_vcuda}
+spack install gcc@${gcc_vcuda}
+spack load gcc@${gcc_vcuda}
+spack compiler find
+spack unload gcc@${gcc_vcuda}
+echo --- Zen2 BLAS+LAPACK
+# Additional development required here due to only partial AMD Rome coverage and
+# spack support for optimized BLAS and LAPACK
+
+spack install amdblis%gcc@${gcc_vnew}
+#spack install blis%gcc@${gcc_vnew}
+# Spack has no amdlibflame package for LAPACK
+#spack install libflame%gcc@${gcc_vnew} # Won't build due to missing python; 
+spack install netlib-lapack%gcc@${gcc_vnew} # Use netlib since it will work and we have wasted enough time already
+
+#spack install openblas%gcc@${gcc_vnew} # Will crash in Performance tests
+
+echo --- Convenience
+spack install git
+echo --- Python setup for NEXUS `date`
+
+spack install py-numpy^openblas%gcc@${gcc_vnew}
+spack install py-mpi4py^openmpi@${ompi_vnew}%gcc@${gcc_vnew} fabrics=auto
```

Comment:
This could be done for the oxygen script as well.  Using that I ended up with with an additional openmpi build triggered by that script since spack defaults to openmpi@3.1.5.

Reply:
Yes, that would be part of the cleanup. These scripts for nitrogen and more specific than those for oxygen. We might be able to use the same script with only a few variations based on the hostname.

Unfortunately there are spack bugs to work around. e.g. py-numpy is built with openblas despite my comments about it being unreliable in QMCPACK because 
```
spack spec py-numpy^amdblis
Input spec
--------------------------------
py-numpy
    ^amdblis

Concretized
--------------------------------
==> Error: Package py-numpy does not depend on amdblis
```
whereas
```
spack spec py-numpy
Input spec
--------------------------------
py-numpy

Concretized
--------------------------------
py-numpy@1.18.1%gcc@9.2.0+blas+lapack arch=linux-rhel8-zen2
    ^openblas@0.3.9%gcc@9.2.0~consistentFPCSR~ilp64+pic+shared threads=none arch=linux-rhel8-zen2
    ^py-cython@0.29.14%gcc@9.2.0 arch=linux-rhel8-zen2
        ^py-setuptools@41.4.0%gcc@9.2.0 arch=linux-rhel8-zen2
            ^python@3.7.6%gcc@9.2.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4~uuid+zlib arch=linux-rhel8-zen2
                ^bzip2@1.0.8%gcc@9.2.0+shared arch=linux-rhel8-zen2
                    ^diffutils@3.7%gcc@9.2.0 arch=linux-rhel8-zen2
                        ^libiconv@1.16%gcc@9.2.0 arch=linux-rhel8-zen2
                ^expat@2.2.9%gcc@9.2.0+libbsd arch=linux-rhel8-zen2
                    ^libbsd@0.10.0%gcc@9.2.0 arch=linux-rhel8-zen2
                ^gdbm@1.18.1%gcc@9.2.0 arch=linux-rhel8-zen2
                    ^readline@8.0%gcc@9.2.0 arch=linux-rhel8-zen2
                        ^ncurses@6.1%gcc@9.2.0~symlinks+termlib arch=linux-rhel8-zen2
                            ^pkgconf@1.6.3%gcc@9.2.0 arch=linux-rhel8-zen2
                ^gettext@0.20.1%gcc@9.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-rhel8-zen2
                    ^libxml2@2.9.9%gcc@9.2.0~python arch=linux-rhel8-zen2
                        ^xz@5.2.4%gcc@9.2.0 arch=linux-rhel8-zen2
                        ^zlib@1.2.11%gcc@9.2.0+optimize+pic+shared arch=linux-rhel8-zen2
                    ^tar@1.32%gcc@9.2.0 arch=linux-rhel8-zen2
                ^libffi@3.2.1%gcc@9.2.0 arch=linux-rhel8-zen2
                ^openssl@1.1.1d%gcc@9.2.0+systemcerts arch=linux-rhel8-zen2
                    ^perl@5.30.1%gcc@9.2.0+cpanm+shared+threads arch=linux-rhel8-zen2
                ^sqlite@3.30.1%gcc@9.2.0~column_metadata+fts~functions~rtree arch=linux-rhel8-zen2
```
i.e. spack py-numpy or amdblis are broken and for now we get an unwanted openblas installation.

An additional hdf5 library is also created somewhere in the python dependencies.

Sadly spack does not try very hard to minimize the amount of freshly installed software: without all the ^openmpi@4 type entries, spack would merrily install openmpi v3.x even when we have a valid instance of v4 installed.

I am hopeful that some of these bugs go away during the next couple of weeks while we wait for the final LLVM 10.0 release and spack support. There has been considerable improvement in the last few weeks.